templates/contribute/naming.md

```diff
@@ -416,3 +416,24 @@ automatically generated unidirectional implications, named `.inj`,
 and there is no intention to change this.
 When such an automatically generated lemma already exists,
 and a bidirectional lemma is needed, it may be named `.inj_iff`.
+
+### Inductive and recursive principles
+
+Inductive/recursive principles are ways to construct data or proofs for all elements of some type `T`,
+by providing ways to construct this data or proof in more constrained specific contexts. 
+These principles generally are phrased to first accept a *motive*,
+which declares what property we are proving or what data we are constructing for all `T`.
+When the motive eliminates into `Prop`, it is an induction principle, and the name should contain
+`induction`. On the other hand, when the motive eliminates into `Sort u` or `Type u`,
+it is a recursive principle, and the name should contain `rec` instead.
+
+Additionally, the name should contain `on` iff in the argument order, the value comes before the constructions.
+
+The following table summarizes these naming conventions.
+
+|motive eliminates into: | `Prop`           | `Sort u` or `Type u`|
+|------------------------|------------------|---------------------|
+| value first            | `T.induction_on` | `T.recOn`           |
+| constructions first    | `T.induction`    | `T.rec`             |
+
+Variation on these names are acceptable when neccessary (e.g. for disambiguation)
```

Comment:
Should we prescrbe whether to use `fooRecOn` or `recFooOn` or `recOnFoo`? Do you know if mathlib is consistent?

Reply:
a quick search using loogle suggests that there doesn't seem to be too much consistency between Init, Mathlib, and the modules between. 
The main takeaway seems to be that both pre- and suffixes are used, but no infixes. (i did notice that `brecOn` seems to be a common name, but i don't know what the `b` signifies, do you know?).